### ISSUE TITLE
Scala.js 1.0 compat: JSExportDescendentObjects has been removed

### ISSUE THREAD

**Issue by SethTisue** [lihaoyi/utest#108]
seen in Scala 2.12 community build, at https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/1408/consoleFull:

```text
[utest] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.7-RC1/project-builds/utest-cc7afacd62432b50d1c26899337c946977b17978/utest/shared/src/main/scala/utest/TestSuite.scala:10: object JSExportDescendentObjects is not a member of package scala.scalajs.js.annotation
[utest] [error] import scala.scalajs.js.annotation.JSExportDescendentObjects
[utest] [error]        ^
[utest] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.7-RC1/project-builds/utest-cc7afacd62432b50d1c26899337c946977b17978/utest/shared/src/main/scala/utest/TestSuite.scala:16: not found: type JSExportDescendentObjects
[utest] [error] @JSExportDescendentObjects
[utest] [error]  ^
[utest] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.7-RC1/project-builds/utest-cc7afacd62432b50d1c26899337c946977b17978/utest/shared/src/main/scala/utest/TestSuite.scala:10: object JSExportDescendentObjects is not a member of package scala.scalajs.js.annotation
[utest] [error] import scala.scalajs.js.annotation.JSExportDescendentObjects
[utest] [error]        ^
[utest] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.7-RC1/project-builds/utest-cc7afacd62432b50d1c26899337c946977b17978/utest/shared/src/main/scala/utest/TestSuite.scala:16: not found: type JSExportDescendentObjects
[utest] [error] @JSExportDescendentObjects
[utest] [error]  ^
[utest] [error] two errors found
```

see https://github.com/scala-js/scala-js/pull/2779

**Comment by sjrd** [comment]
See also the release notes of Scala.js 0.6.15: https://www.scala-js.org/news/2017/03/21/announcing-scalajs-0.6.15/

The fix should be as simple as swapping `@JSExportDescendentObjects` for `@scala.scalajs.reflect.annotation.EnableReflectiveInstantiation`.

**Comment by sjrd** [comment]
Fix available in #109.
